auth/auth.go

```diff
@@ -98,40 +78,45 @@ type signer interface {
 //
 // This function can only be invoked from within the SDK. Client applications should access the
 // Auth service through firebase.App.
-func NewClient(ctx context.Context, c *internal.AuthConfig) (*Client, error) {
+func NewClient(ctx context.Context, conf *internal.AuthConfig) (*Client, error) {
 	var (
-		err   error
-		email string
-		pk    *rsa.PrivateKey
+		signer cryptoSigner
+		err    error
 	)
-	if c.Creds != nil && len(c.Creds.JSON) > 0 {
-		var svcAcct struct {
-			ClientEmail string `json:"client_email"`
-			PrivateKey  string `json:"private_key"`
-		}
-		if err := json.Unmarshal(c.Creds.JSON, &svcAcct); err != nil {
+	// Initialize a signer by following the go/firebase-admin-sign protocol.
+	if conf.Creds != nil && len(conf.Creds.JSON) > 0 {
```

Comment:
The description of this PR says that a nil conf will be allowed, but this would crash in that case. I think this is probably the right behavior, and the description is wrong. But it would be good to sync them up.

Reply:
The conf passed to `NewApp()` can be nil. `NewAuth()` is an internal API, and always get called with a non-nil conf.

Reply:
Assuming by `NewAuth()` you meant `auth.NewClient()`, then SGTM.